jujugui/static/gui/src/app/init/component-renderers-mixin.js

```diff
@@ -0,0 +1,1080 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const yui = window.yui;
+
+const ComponentRenderersMixin = (superclass) => class extends superclass {
```

Comment:
I've added the doc block.

I totally agree that splitting the code out like this does little to clean up those types of things however it's a little out of scope for this work. Something for a follow-up